### PR TITLE
Refactor docs organization

### PR DESCRIPTION
**Pull request by bridgetkromhout**
First pass of reorganizing docs based on https://github.com/openservicemesh/osm-docs/issues/84#issuecomment-858111450.

### CODE REVIEW

File: content/docs/concepts_features/_index.md [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ---
-title: "Design & Concepts"
+title: "Concepts and Features"
 description: "Detailed design and architecture of Open Service Mesh."
 type: docs
-aliases: ["design & concepts"]
 weight: 3
```
Comment (zr-msft) [inline content/docs/concepts_features/_index.md:5]:
```suggestion
weight: 4
```

File: content/docs/tasks/_index.md [hunk 2]
```diff
@@ -1,7 +1,6 @@
 ---
-title: "Tasks & Usage"
+title: "Tasks"
 description: "Learn how to do common activities with Open Service Mesh (OSM)"
 type: docs
-aliases: ["tasks"]
 weight: 4
```
Comment (zr-msft) [inline content/docs/tasks/_index.md:5]:
```suggestion
weight: 3
```

**Review by zr-msft — COMMENTED** [review]
I suggest we swap concepts/tasks in the toc order

**Comment by bridgetkromhout** [comment]
> I suggest we swap concepts/tasks in the toc order

@zr-msft Done.

**Comment by zr-msft** [comment]
LGTM!

approved

**Review by zr-msft — APPROVED** [review]

**Pull request merged**